### ISSUE TITLE
"gcc: error: @GLOBUS_CFLAGS@: No such file or directory" when "make" gct

### ISSUE THREAD

**Issue by iWangJiaxiang** [gridcf/gct#79]
Hello, I encountered an fatal error as below when I "make" GCT from the source code of latest release [GCT 6.2.20190226 (maintenance release)](https://github.com/gridcf/gct/releases/tag/v6.2.20190226)

     make[4]: Entering directory /home/globus/gct-6.2.1550507116/gsi_openssh/source/openbsd-compat
     make[4]: Nothing to be done for all.
     make[4]: Leaving directory \/home/globus/gct-6.2.1550507116/gsi_openssh/source/openbsd-compat\'
     gcc -g -O2 -Wall -Wpointer-arith -Wuninitialized -Wsign-compare -Wformat-security -Wsizeof-pointer-memaccess -Wno-pointer-sign -Wno-unused-result -fno-strict-aliasing -D_FORTIFY_SOURCE=2 -ftrapv -fno-builtin-memset -fstack-protector-strong -fPIE  @GLOBUS_CFLAGS@ -I. -I.  -D_XOPEN_SOURCE=600 -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_PATH_SSH_ASKPASS_DEFAULT=\"/home/globus/globus/libexec/ssh-askpass\" -DGSISSHDIR="\"/home/globus/globus/etc/gsissh\"" -D_PATH_SSH_PIDDIR=\"/var/run\" -D_PATH_PRIVSEP_CHROOT_DIR=\"/var/empty\" -DHAVE_CONFIG_H -c ssh_api.c -o ssh_api.o
     gcc: error: @GLOBUS_CFLAGS@: No such file or directory
     make[3]: *** [ssh_api.o] Error 1
     make[3]: Leaving directory /home/globus/gct-6.2.1550507116/gsi_openssh/source'
     make[2]: *** [gsi-openssh-stamp] Error 1
     make[2]: Leaving directory /home/globus/gct-6.2.1550507116'
     make[1]: *** [all-recursive] Error 1
     make[1]: Leaving directory /home/globus/gct-6.2.1550507116'
     make: *** [all] Error 2

Please pay attention to line 5:

    gcc: error: @GLOBUS_CFLAGS@: No such file or directory

Below is my "./configure" command:

    ./configure --prefix=/home/globus/globus --enable-myproxy --enable-gram5-server --pdfdir=/home/globus/ --with-openssl 2>&1 | tee log.txt

I run all commands in user "globus" but I got the error above. I tried the same on another server but the error occurred still.

I also searched "GLOBUS_CFLAGS" in this GitHub Repo but got no word totally same.

Attached with my server environment:

\ |machine 1|VM 1
:-:|:-:|:-:
OS|RedHat 5.3|CentOS 7.6
Arch|x86_64|x86_64
Compiler|gcc|gcc
Openssl Version|1.1.1b|1.1.1b

Could you please help me work out?

**Comment by iWangJiaxiang** [comment]
I tried to compile the official globus toolkit6.0 from the source with same parameters and succeeded. It seems like a bug in GCT release?

**Comment by iWangJiaxiang** [comment]
I tried with the previous release [v6.2.20181115](https://github.com/gridcf/gct/releases/tag/v6.2.20181115) and everything is fine. So the problem might be an issue in the latest release.

Below is the correct output of line that caused error in the lastest release.

![image](https://user-images.githubusercontent.com/19409802/56492786-74d4b780-651f-11e9-889a-79521b411959.png)

    gcc -g -O2  -Wall -Wpointer-arith -Wsign-compare -Wformat-security -Wsizeof-pointer-memaccess -Wno-pointer-sign -Wno-unused-result -fno-strict-aliasing -D_FORTIFY_SOURCE=2 -ftrapv -fno-builtin-memset -fstack-protector-strong -fPIE  -g -O2  -g -O2 -pthread -I/home/globus/gct-6.2.1541705016/common/source -I/home/globus/gct-6.2.1541705016/common/source/library -I/home/globus/gct-6.2.1541705016/xio/src -I/home/globus/gct-6.2.1541705016/xio/src/builtins/file -I/home/globus/gct-6.2.1541705016/xio/src/builtins/http -I/home/globus/gct-6.2.1541705016/xio/src/builtins/mode_e -I/home/globus/gct-6.2.1541705016/xio/src/builtins/ordering -I/home/globus/gct-6.2.1541705016/xio/src/builtins/queue -I/home/globus/gct-6.2.1541705016/xio/src/builtins/tcp -I/home/globus/gct-6.2.1541705016/xio/src/builtins/telnet -I/home/globus/gct-6.2.1541705016/xio/src/builtins/udp -I/home/globus/gct-6.2.1541705016/gsi/openssl_error/source -I/home/globus/gct-6.2.1541705016/gsi/openssl_error/source/library -I/home/globus/gct-6.2.1541705016/gsi/proxy/proxy_ssl/source -I/home/globus/gct-6.2.1541705016/gsi/proxy/proxy_ssl/source/library -I/home/globus/gct-6.2.1541705016/gsi/openssl_module/source/library -I/home/globus/gct-6.2.1541705016/gsi/cert_utils/source/library -I/home/globus/gct-6.2.1541705016/gsi/sysconfig/source/library -I/home/globus/gct-6.2.1541705016/callout/source/library -I/home/globus/gct-6.2.1541705016/gsi/callback/source/library -I/home/globus/gct-6.2.1541705016/gsi/credential/source/library -I/home/globus/gct-6.2.1541705016/gsi/proxy/proxy_core/source/library -I/home/globus/gct-6.2.1541705016/gsi/gssapi/source/library -I/home/globus/gct-6.2.1541705016/gsi/gss_assist/source -I/home/globus/gct-6.2.1541705016/usage/c/sender/source -I. -I.  -D_XOPEN_SOURCE=600 -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_PATH_SSH_ASKPASS_DEFAULT=\"/home/globus/globus/libexec/ssh-askpass\" -DGSISSHDIR="\"/home/globus/globus/etc/gsissh\"" -D_PATH_SSH_PIDDIR=\"/var/run\" -D_PATH_PRIVSEP_CHROOT_DIR=\"/var/empty\" -DHAVE_CONFIG_H -c sshkey.c -o sshkey.o

And below is the error output:

![image](https://user-images.githubusercontent.com/19409802/56492744-579fe900-651f-11e9-81fb-63febd22c49e.png)

    gcc -g -O2 -Wall -Wpointer-arith -Wuninitialized -Wsign-compare -Wformat-security -Wsizeof-pointer-memaccess -Wno-pointer-sign -Wno-unused-result -fno-strict-aliasing -D_FORTIFY_SOURCE=2 -ftrapv -fno-builtin-memset -fstack-protector-strong -fPIE  @GLOBUS_CFLAGS@ -I. -I.  -D_XOPEN_SOURCE=600 -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_PATH_SSH_ASKPASS_DEFAULT=\"/home/globus/globus/libexec/ssh-askpass\" -DGSISSHDIR="\"/home/globus/globus/etc/gsissh\"" -D_PATH_SSH_PIDDIR=\"/var/run\" -D_PATH_PRIVSEP_CHROOT_DIR=\"/var/empty\" -DHAVE_CONFIG_H -c ssh_api.c -o ssh_api.o
    gcc: error: @GLOBUS_CFLAGS@: No such file or directory
    make[3]: *** [ssh_api.o] Error 1
    make[3]: Leaving directory `/home/globus/gct-6.2.1550507116/gsi_openssh/source'
    make[2]: *** [gsi-openssh-stamp] Error 1
    make[2]: Leaving directory `/home/globus/gct-6.2.1550507116'
    make[1]: *** [all-recursive] Error 1
    make[1]: Leaving directory `/home/globus/gct-6.2.1550507116'
    make: *** [all] Error 2

I have checked the comparison between the two releases and not found anything related for now. I would continue to check when not busy.

**Comment by ellert** [comment]
Thank you for reporting this.
This is fixed in git master (commits 7c0d37a29a and d4b51f4fc3).
The workaround for building using the gct-6.2.1550507116.tar.gz source tarball is:
```
tar -z -x -f gct-6.2.1550507116.tar.gz
cd gct-6.2.1550507116
( cd gsi_openssh/source; aclocal; autoheader; autoconf )
./configure
```
I'll leave this issue open until there is a new release not affected by the issue.

**Comment by fscheiner** [comment]
Included in GCT v6.2.20190906.
